sechub-wrapper-prepare/src/main/java/com/mercedesbenz/sechub/wrapper/prepare/moduls/PrepareWrapperGIT.java

```diff
@@ -0,0 +1,200 @@
+            throw new IOException("Error while cloning repository: " + gitContext.location + "GIT Error: " + e.getMessage(), e);
+        }
+        waitForProcessToFinish();
+    }
+
+    public void cleanGitDirectory(String pdsPrepareUploadFolderDirectory) throws IOException {
+        List<String> commands = new ArrayList<>();
+        // removes recursively all git files from a directory
+        // ( find . -type d -name ".git" && find . -name ".gitignore" && find . -name
+        // ".gitmodules" ) | xargs rm -rf
+
+        File uploadDir;
+        uploadDir = Paths.get(pdsPrepareUploadFolderDirectory).toAbsolutePath().toFile();
+
+        commands.add("bash");
+        commands.add("-c");
+        commands.add("( find . -type d -name .git && find . -name .gitignore && find . -name .gitattributes ) | xargs rm -rf");
+
+        ProcessBuilder builder = new ProcessBuilder(commands);
+        builder.directory(uploadDir);
+        try {
+            process = processAdapterFactory.startProcess(builder);
+        } catch (Exception e) {
+            LOG.error("Error while cleaning git directory: " + pdsPrepareUploadFolderDirectory, e);
+            throw new IOException("Error while cleaning git directory: " + pdsPrepareUploadFolderDirectory, e);
+        }
+        waitForProcessToFinish();
+    }
+
+    String getRepositoryURL(String location) {
+        LOG.debug("Prepare location string for private repository");
+        String preFix = "https://" + "$" + PDS_PREPARE_CREDENTIAL_USERNAME + ":" + "$" + PDS_PREPARE_CREDENTIAL_PASSWORD + "@";
+
+        String repositoryURL;
+        if (location.contains("https://")) {
+            repositoryURL = location.replace("https://", preFix);
+        } else if (location.contains("git@")) {
+            location = location.replace(":", "/");
+            repositoryURL = location.replace("git@", preFix);
+        } else if (location.contains("http://")) {
+            repositoryURL = location.replace("http://", preFix);
+        } else if (location.contains("git://")) {
+            repositoryURL = location.replace("git://", preFix);
+        } else {
+            repositoryURL = preFix + location;
+        }
+        return repositoryURL;
+    }
+
+    private void waitForProcessToFinish() {
+
+        /* watching */
+        String watcherThreadName = "PDSPrepareWrapper-git-watcher";
+
+        LOG.debug("Start watcher thread: {}", watcherThreadName);
+
+        StreamDataRefreshRequestWatcherRunnable streamDatawatcherRunnable = new StreamDataRefreshRequestWatcherRunnable();
+        Thread streamDataWatcherThread = new Thread(streamDatawatcherRunnable);
+        streamDataWatcherThread.setName(watcherThreadName);
+        streamDataWatcherThread.start();
+
+        /* waiting for process */
+        LOG.debug("Wait for GIT wrapper to finish process.");
+
+        boolean exitDoneInTime = false;
+        try {
+            exitDoneInTime = process.waitFor(minutesToWaitForResult, TimeUnit.MINUTES);
```

Comment:
Again, be aware here - maybe setup with seconds (and ensured to be smaller than script timeout?) would be better?